converter/shieldhit/parser.py

```diff
@@ -207,6 +226,37 @@ def _get_detector_by_uuid(self, detect_uuid: str) -> Optional[str]:
 
         raise ValueError(f"No detector with uuid {detect_uuid}")
 
+    def _parse_quantity_settings(self, quantity_dict: dict) -> dict or None:
+        """Parses settings from the input json into the quantity settings property"""
+
+        def create_name_from_settings() -> str:
+            """Create a name for the quantity from its settings."""
+            if re.search(r'^Quantity(_\d*)?$', quantity_dict['name']):
+                return f"""{'Absolute_'
+                    if 'primaries' in quantity_dict
+                    else 'Rescaled_'
+                    if 'rescale' in quantity_dict
+                    else ''}{quantity_dict['name']}{('_to_'+quantity_dict['medium'])
+                    if 'medium' in quantity_dict
+                    else ('_to_'+self._get_material_by_uuid(quantity_dict['materialUuid']).sanitized_name)
+                    if 'materialUuid' in quantity_dict
+                    else ''
+                }"""
```

Comment:
```suggestion
                part1 = 'Absolute_' if 'primaries' in quantity_dict else 'Rescaled_'
                return f"""{part1
                    if 'rescale' in quantity_dict
                    else ''}{quantity_dict['name']}{('_to_'+quantity_dict['medium'])
                    if 'medium' in quantity_dict
                    else ('_to_'+self._get_material_by_uuid(quantity_dict['materialUuid']).sanitized_name)
                    if 'materialUuid' in quantity_dict
                    else ''
                }"""
```

Can you split this complex f-string into more pieces of the code, introducing some meaningful variables ?